lib/puppet_x/coi/jboss/factsrefresher.rb

```diff
@@ -0,0 +1,10 @@
+require_relative 'facts'
+
+class Puppet_X::Coi::Jboss::FactsRefresher
+  class << self
+
+    def refresh_facts
+      Puppet_X::Coi::Jboss::Facts::define_fullconfig_fact
```

Comment:
To properly refresh those fact you need to remove previous implementation and value from them. Setting want help :sob: !fix